pages/fifo/[url].tsx

```diff
@@ -75,63 +74,98 @@ const eventIsOver = () => {
 }
 
 const fifo = (): JSX.Element => {
-  const [isOver, setIsOver] = useState(false)
+  const [eventStart, setEventStart] = useState(false)
   const [eventData, setEventData] = useState<IeventData | null>(null)
+  const [eventOver, setEventOver] = useState(false)
+  const [distance, setDistance] = useState(0)
   const router = useRouter()
-  let timer: NodeJS.Timer
-
-  const initEvent = () => {
-    //api 로직
-    //이벤트 종료 여부 체크
-    if (MOCK_DATA.eventProgressStatus === 'IsOver') {
-      setIsOver(true)
-    } else {
-      setIsOver(false)
-    }
-    setEventData(() => MOCK_DATA)
-  }
+  const { updateUser } = useUserContext()
+
+  // 선물 수령 API
+  const handleGiftReceipt = useCallback(
+    async (e: React.MouseEvent<HTMLButtonElement>) => {
+      if (eventStart && eventData) {
+        const eventId = eventData.eventId
+        const giftId = parseInt((e.target as HTMLElement).id, 10) // 카테고리 ID
+        const memberId = eventData.member.id
+        const res = await postGiftReceipt({ eventId, giftId, memberId })
+        if (Array.isArray(res)) {
+          //res = ['E002', '이미 참여한 이벤트입니다.']
+          const errorMessage = res[1]
+          alert(errorMessage)
+        } else {
+          // 선물 수령 완료 이후 /gift/res.id로 이동
+          alert('선물 겟!')
+          // router.push('/gift/${res.id}')
+        }
+      } else {
+        alert('지금은 선물을 받을 수 없어요!')
+      }
+    },
+    [eventStart],
+  )
 
-  const onButtonClick = () => {
-    if (isOver) {
-      // if '선물을 받은 사람인지 아닌지 체크!'
-      // 선물 받기 전에 수량 체크 한번 더 해야함.
-      // 수량 체크 해서 이상없으면 받고, 있으면 못받음! + 데이터 한번 더 받아야함.
-      console.log('서버에 해당 선물 당첨자 저장')
-      alert('선물 겟')
-    } else {
-      alert('지금은 선물을 받을 수 없어요!')
+  // setInterval Clear 함수
+  const clear = useInterval(() => {
+    checkRemaining()
+  }, 1000)
+
+  // 남아있는 시간 체크 함수 (현재 시간 - 이벤트 시작 시간)
+  const checkRemaining = () => {
+    if (eventData?.startAt) {
+      const now = new Date()
+      const eventStartAt = new Date(eventData?.startAt)
+      const distance = Number(eventStartAt) - Number(now)
+      setDistance(distance)
     }
   }
```

Comment:
```suggestion
  const checkRemaining = useCallback(() => {
    if (eventData?.startAt) {
      const now = new Date()
      const eventStartAt = new Date(eventData.startAt) // if문에서 eventData 체킹이 있어서 여기선 안해줘도 될 것 같아요
      const distance = Number(eventStartAt) - Number(now)
      setDistance(distance)
    }
  }, [eventData])
```
eventData에 의존성을 주는 것도 좋을 것 같습니다!

Reply:
오 피드백 반영하였습니다.
타이머 정상 동작 확인하였습니다.